nautobot/extras/models/groups.py

```diff
@@ -528,45 +289,117 @@ def get_filter_fields(self):
+        else:
+            for obj in value:
+                if not isinstance(obj, self.model):
+                    raise TypeError(f"{obj} is not a {self.model._meta.label_lower}")
+            to_remove = []
+            for member in self.members:
+                if member not in value:
+                    to_remove.append(member)
+            self._remove_members(to_remove)
+            to_add = []
+            members = self.members
+            for candidate in value:
+                if candidate not in members:
+                    to_add.append(candidate)
+            self._add_members(to_add)
 
-        filterset = self.filterset_class(self.filter, model.objects.all())
-        if not filterset.is_valid():
-            logger.warning('Filter for DynamicGroup "%s" is not valid', self)
-            return model.objects.none()
+        return self.members
 
-        qs = filterset.qs
+    def add_members(self, objects_to_add):
+        """Add the given list or QuerySet of objects to this staticly defined group."""
+        if self.group_type != DynamicGroupTypeChoices.TYPE_STATIC:
+            raise ValidationError(f"Group {self} is not staticly defined, adding members directly is not permitted.")
+        return self._add_members(objects_to_add)
 
-        # Make sure that this instance can't be a member of its own group.
-        if self.present_in_database and model == self.__class__:
-            qs = qs.exclude(pk=self.pk)
+    def _add_members(self, objects_to_add):
+        """Internal API for adding the given list or QuerySet of objects to the cached/static members of this group."""
+        if isinstance(objects_to_add, models.QuerySet):
+            if objects_to_add.model != self.model:
+                raise TypeError(f"QuerySet does not contain {self.model._meta.label_lower} objects")
+        else:
+            for obj in objects_to_add:
+                if not isinstance(obj, self.model):
+                    raise TypeError(f"{obj} is not a {self.model._meta.label_lower}")
 
-        return qs
+        if self.group_type == DynamicGroupTypeChoices.TYPE_STATIC:
+            for obj in objects_to_add:
+                # We don't use `.bulk_create()` currently because we want change logging for these creates.
+                # Might be a good future performance improvement though.
+                StaticGroupAssociation.all_objects.get_or_create(
+                    dynamic_group=self, associated_object_type=self.content_type, associated_object_id=obj.pk
+                )
+        else:
+            # Cached/hidden static group associations, so we can use bulk-create to bypass change logging.
+            sgas = [
+                StaticGroupAssociation(
+                    dynamic_group=self, associated_object_type=self.content_type, associated_object_id=obj.pk
+                )
+                for obj in objects_to_add
+            ]
+            StaticGroupAssociation.all_objects.bulk_create(sgas)
```

Comment:
Hmm... does this risk IntegrityErrors if it creates duplicate records? Do we need to compare `objects_to_add` against existing records first?